lms/services/application_instance.py

```diff
@@ -54,17 +54,15 @@ class ApplicationInstanceService:
     def __init__(
         self,
         db,
-        request,
         aes_service: AESService,
         organization_service: OrganizationService,
     ):
         self._db = db
-        self._request = request
         self._aes_service = aes_service
         self._organization_service = organization_service
 
     @lru_cache(maxsize=1)
-    def get_current(self) -> ApplicationInstance:
+    def get_for_launch(self, id_) -> ApplicationInstance:
```

Comment:
This is probably the biggest change.

Before `get_current` was based on `lti_user.application_instance_id

That was not clear from the outside and we needed the `lru_cache` to avoid making the query all the time.


Now we get use this method while building LTIUser just once.